src/trajectory_generator.cpp

```diff
@@ -47,11 +47,12 @@ TrajectoryGenerator::TrajectoryGenerator(uint num_dof, double timestep, double d
 {
   // Upsample if num. waypoints would be short. Helps with accuracy
   upsample();
+  double max_num_waypoints = max_duration / upsampled_timestep_;
```

Comment:
Sorry, but I think this should be a size_t. `kMaxNumWaypointsFullTrajectory` that it replaced was a size_t

Reply:
Thinking more about this, it would be wise to always round up so the result is predictable, I think:

```suggestion
  size_t max_num_waypoints = (size_t) std::ceil( max_duration / upsampled_timestep_ );
```